src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -318,7 +318,10 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary = False):
                     jobState = "jobfailed"
                 else:
                     jobState = newstate
-                    
+                
+                # there is race condition updating couch record location and job is completed0
+                # for the fast fail job, it could miss the location update
+                job["location"] = job["fwjr"].getSiteName() or job.get("location", "UnKnown")     
```

Comment:
Job can't have multiple location at this point although default value for getSiteName() is {}.
siteName comes from siteConfig, AFAIK, it is just one site.
https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/WMRuntime/Bootstrap.py#L235

What could happen is that job might failed with no fwjr. in that case it will show the location which agent selected. I am not sure whether that is desired or we should set unknown. I didn't set unknown because in normal case when JobTracker tracks down the running job it would have correct site even though fwjr doesn't exist.